### ISSUE TITLE
Mac OSX Homebrew installation directory issue

### ISSUE THREAD

**Issue by sanak** [pgRouting/pgrouting#345]
I am trying Mac OSX Homebrew environment, and noticed that
pgrouting installation directory is always inside PostgreSQL directory,
regardress of `CMAKE_INSTALL_PREFIX` value.

The following is Homebrew pgrouting installation log.
<pre>
MBP2012:Homebrew sanak$ brew install -v pgrouting
==> Downloading https://github.com/pgRouting/pgrouting/archive/v2.0.0.tar.gz
Already downloaded: /Library/Caches/Homebrew/pgrouting-2.0.0.tar.gz
==> Verifying pgrouting-2.0.0.tar.gz checksum
tar xf /Library/Caches/Homebrew/pgrouting-2.0.0.tar.gz
==> Patching
patching file src/astar/src/astar.h
Hunk #1 succeeded at 21 with fuzz 1.
patching file src/dijkstra/src/dijkstra.h
Hunk #1 succeeded at 22 with fuzz 1.
patching file src/driving_distance/src/drivedist.h
==> cmake -DWITH_DD=ON .. -DCMAKE_C_FLAGS_RELEASE= -DCMAKE_CXX_FLAGS_RELEASE= -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/pgrouting/2.0.0 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev
-- The C compiler identification is AppleClang 6.1.0.6020053
-- The CXX compiler identification is AppleClang 6.1.0.6020053
-- Check for working C compiler: /usr/local/Library/ENV/4.3/clang
-- Check for working C compiler: /usr/local/Library/ENV/4.3/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/Library/ENV/4.3/clang++
-- Check for working CXX compiler: /usr/local/Library/ENV/4.3/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- PGROUTING_GIT_TAG: pgrouting-2.0.0
-- PGROUTING_GIT_BUILD: 0
-- PGROUTING_GIT_HASH: f26831f
-- PGROUTING_GIT_BRANCH: master
-- PGROUTING_VERSION_REVISION_NAME: f26831f master
-- PGROUTING_VERSION_REVISION_HASH: f26831f
-- UNIX=1
-- WIN32=
-- POSTGRESQL_EXECUTABLE is /usr/local/bin/postgres
-- POSTGRESQL_PG_CONFIG is /usr/local/bin/pg_config
-- POSTGRESQL_INCLUDE_DIR: /usr/local/Cellar/postgresql/9.4.2/include/server
-- POSTGRESQL_LIBRARIES: /usr/local/Cellar/postgresql/9.4.2/lib
-- Boost version: 1.58.0
-- Boost headers were found here: /usr/local/include
-- POSTGRESQL_VERSION_STRING is PostgreSQL 9.4.2
-- POSTGRESQL_VERSION is 9.4.2
-- PGSQL_VERSION is 94
-- Install directory for libraries is set to /usr/local/Cellar/postgresql/9.4.2/lib
-- Contrib directory for SQL files is set to /usr/local/Cellar/postgresql/9.4.2/share/postgresql/contrib/pgrouting-2.0
-- Extension directory for SQL files is set to /usr/local/Cellar/postgresql/9.4.2/share/postgresql/extension
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   thread
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   thread
--   system
-- CGAL_INCLUDE_DIR=/usr/local/include
-- CGAL_LIBRARIES=/usr/local/lib/libCGAL.dylib
-- BOOST_THREAD_LIBRARIES=/usr/local/lib/libboost_thread-mt.dylib;/usr/local/lib/libboost_system-mt.dylib
-- GMP_LIBRARIES=/usr/local/lib/libgmp.dylib
-- Found CGAL: /usr/local/include, /usr/local/lib/libCGAL.dylib, /usr/local/lib/libboost_thread-mt.dylib;/usr/local/lib/libboost_system-mt.dylib, /usr/local/lib/libgmp.dylib
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/pgrouting20150602-4515-aa7ml5/pgrouting-2.0.0/build
==> make install
:
:
Install the project...
/usr/local/Cellar/cmake/3.2.2/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /usr/local/Cellar/postgresql/9.4.2/lib/librouting_bd.so
-- Installing: /usr/local/Cellar/postgresql/9.4.2/lib/librouting.so
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/extension/pgrouting--2.0.0.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/extension/pgrouting.control
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/extension/pgrouting_legacy.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/contrib/pgrouting-2.0/pgrouting.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/contrib/pgrouting-2.0/pgrouting.control
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/contrib/pgrouting-2.0/pgrouting_legacy.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/extension/pgrouting_dd_legacy.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.2/share/postgresql/contrib/pgrouting-2.0/pgrouting_dd_legacy.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.2/lib/librouting_tsp.so
-- Installing: /usr/local/Cellar/postgresql/9.4.2/lib/librouting_ksp.so
-- Installing: /usr/local/Cellar/postgresql/9.4.2/lib/librouting_dd.so
==> Cleaning
==> Finishing up
==> Summary
🍺  /usr/local/Cellar/pgrouting/2.0.0: 3 files, 28K, built in 14 seconds
</pre>

As a result, Homebrew pgrouting can't manage installed modules, although postgis and sfcgal can manage those ones, and therefore, pgrouting can't be uninstalled.
<pre>
$ brew list pgrouting
### Nothing ###
$ brew list sfcgal
/usr/local/Cellar/sfcgal/1.0.5_2/bin/sfcgal-config
/usr/local/Cellar/sfcgal/1.0.5_2/include/SFCGAL/ (105 files)
/usr/local/Cellar/sfcgal/1.0.5_2/lib/libSFCGAL.1.0.5.dylib
/usr/local/Cellar/sfcgal/1.0.5_2/lib/ (2 other files)
$ brew list postgis
/usr/local/Cellar/postgis/2.1.7_1/bin/create_undef.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/pgsql2shp
/usr/local/Cellar/postgis/2.1.7_1/bin/postgis_proc_upgrade.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/postgis_restore.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/profile_intersects.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/raster2pgsql
/usr/local/Cellar/postgis/2.1.7_1/bin/shp2pgsql
/usr/local/Cellar/postgis/2.1.7_1/bin/test_estimation.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/test_geography_estimation.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/test_geography_joinestimation.pl
/usr/local/Cellar/postgis/2.1.7_1/bin/test_joinestimation.pl
/usr/local/Cellar/postgis/2.1.7_1/include/liblwgeom.h
/usr/local/Cellar/postgis/2.1.7_1/lib/liblwgeom-2.1.7.dylib
/usr/local/Cellar/postgis/2.1.7_1/lib/ (2 other files)
/usr/local/Cellar/postgis/2.1.7_1/share/man/ (2 files)
/usr/local/Cellar/postgis/2.1.7_1/share/postgis/ (23 files)
</pre>

I am not sure whether this is a pgrouting design, but I think that it is better to check
other similar libraries(postgis/sfcgal) behavior of each platform.

**Comment by cvvergara** [comment]
@sanak Does this happen also on develop_2_1_0 ?

**Comment by sanak** [comment]
@cvvergara Sorry, I missed your comment. I will try to check it.

**Comment by sanak** [comment]
> Does this happen also on develop_2_1_0 ?

Yes, this seems to happen on the latest "develop" branch.

`cmake -DCMAKE_INSTALL_PREFIX=/tmp/pgrouting ../` results:
<pre>
:
-- Install directory for libraries is set to /usr/local/Cellar/postgresql/9.4.4/lib
-- Contrib directory for SQL files is set to /usr/local/Cellar/postgresql/9.4.4/share/postgresql/contrib/pgrouting-2.1
-- Extension directory for SQL files is set to /usr/local/Cellar/postgresql/9.4.4/share/postgresql/extension
:
</pre>
`make install` results:
<pre>
:
Install the project...
-- Install configuration: ""
-- Installing: /usr/local/Cellar/postgresql/9.4.4/lib/librouting-2.1.so
-- Installing: /usr/local/Cellar/postgresql/9.4.4/share/postgresql/extension/pgrouting--2.1.0.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.4/share/postgresql/extension/pgrouting.control
-- Installing: /usr/local/Cellar/postgresql/9.4.4/share/postgresql/contrib/pgrouting-2.1/pgrouting.sql
-- Installing: /usr/local/Cellar/postgresql/9.4.4/share/postgresql/contrib/pgrouting-2.1/pgrouting.control
</pre>

I think that this is Mac OSX homebrew specific requirement, so next release (2.2.0) is enough to support it.

**Comment by cvvergara** [comment]
I dont know how to fix that, I am leaving this issue for next release.

**Comment by sanak** [comment]
Okay, then I will check it at that time.

**Comment by cvvergara** [comment]
@sanak 
I dont know if this #370  might have fixed this problem

**Comment by titanofold** [comment]
I don't think this is a bug in pgrouting.

The only files produced by pgrouting are the module and the SQL, both of which must to be placed in PostgreSQL's lib/ and/or extension/. You're list for PostGIS is incomplete, too. There are dozens of files that PostGIS installs to PostgreSQL's lib/ and extension/ that brews isn't listing.

So, this doesn't look so much like a bug with pgrouting, but with brew's inability to cope with PostgreSQL's extension system.

In short, pgrouting is installing correctly. Any other way would be wrong.

**Comment by cvvergara** [comment]
I am going to leave this issue open as discussion with no milestone, so we keep don't forget about it.

**Comment by sanak** [comment]
@titanofold 
oka

**Comment by livemixlove** [comment]
Any updates on this? 

I run: brew install postgis
Then in psql on my database: CREATE EXTENSION postgis; 

and I get:

`ERROR:  could not open extension control file "/usr/local/Cellar/postgresql/9.4.2/share/postgresql/extension/postgis.control": No such file or directory`

Is there a workaround?

**Comment by cvvergara** [comment]
@robe2 might be able to answer this but she is in a trip.
We already revised installation instructions for next release here:
http://docs.pgrouting.org/dev/doc/src/installation/installation.html
there is an install guide for postGIS
Can you have a look to that guide?.

**Comment by sanak** [comment]
@livemixlove 
It is not pgrouting but postgis.
Could you try the following ?
<pre>
brew update
brew uninstall postgis
brew uninstall pgrouting
brew install postgis
brew install pgrouting
</pre>

**Comment by livemixlove** [comment]
Ok, so I figured it out.  I did not notice that Brew had updated my postgres install, and so I had not restarted the database server.  My server was running 9.4 and my new install was 9.5.  Postgis had been installed linked with the 9.5 version.  Once I restarted the server pointing using the new 9.5, it worked.

**Comment by sanak** [comment]
@livemixlove 
Okay, thanks for confirmation.

About original issue, I confirmed that recent homebrew package is fixed already, so I will close this tickt.
